### PR TITLE
pull request #6 from tenyiting/chapter_6_modeling_user

### PR DESCRIPTION
**Pull request by tenyiting**
chapter_6_modeling_user

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```
Comment (huongnt-2545) [inline app/models/user.rb]: line 7, 8
```suggestion
  validates :email, presence: true, uniqueness: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```

File: test/fixtures/users.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/users.yml]: delete this file, it's no need

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (huongnt-2545) [inline test/models/user_test.rb]: delete this file

File: test/fixtures/users.yml (hunk 2 above)
Comment (tenyiting) [reply test/fixtures/users.yml]: Screenshot from 2021-09-01 21-03-55

File: test/models/user_test.rb (hunk 3 above)
Comment (tenyiting) [reply test/models/user_test.rb]: Screenshot from 2021-09-01 21-06-20

File: app/models/user.rb (hunk 1 above)
Comment (tenyiting) [reply app/models/user.rb]: Screenshot from 2021-09-01 21-12-10

**Comment by tenyiting** [comment]
ready

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true, uniqueness: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```
Comment (ngocvt-0484) [inline app/models/user.rb]: wrong indent

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  has_secure_password
+  private
+
+  def downcase_email
+    self.email.downcase!
+  end
+
+
```
Comment (ngocvt-0484) [inline app/models/user.rb]: remove blank line

**Comment by tenyiting** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**